File: android/src/main/java/com/audiocontext/nodes/AudioDestinationNode.kt
package com.audiocontext.nodes

import com.audiocontext.context.BaseAudioContext
import com.audiocontext.parameters.PlaybackParameters


class AudioDestinationNode(context: BaseAudioContext): AudioNode(context) {
  override val numberOfInputs = Float.POSITIVE_INFINITY.toInt()
  override val numberOfOutputs = 0

  private fun setVolumeAndPanning(playbackParameters: PlaybackParameters) {
    playbackParameters.audioTrack.setStereoVolume(playbackParameters.leftPan.toFloat(), playbackParameters.rightPan.toFloat())
  }

  override fun process(playbackParameters: PlaybackParameters) {
    setVolumeAndPanning(playbackParameters)
    playbackParameters.audioTrack.write(playbackParameters.buffer, 0, playbackParameters.buffer.size)
  }
}

File: packages/react-native-audio-api/src/core/AudioDestinationNode.ts
import { IAudioDestinationNode } from '../interfaces';
import AudioNode from './AudioNode';
import BaseAudioContext from './BaseAudioContext';

export default class AudioDestinationNode extends AudioNode {
  constructor(context: BaseAudioContext, destination: IAudioDestinationNode) {
    super(context, destination);
  }
}
